editor/block-right-menu/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from 'i18n';
+import { IconButton } from 'components';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function BlockRightMenu( { onDelete } ) {
+	return (
+		<div className="editor-block-right-menu">
+			<IconButton
+				className="editor-block-right-menu__control"
+				onClick={ onDelete }
+				icon="trash"
+				label={ __( 'Delete the block' ) }
```

Comment:
We use the position for the block mover. I can do the same `Delete the block at the position %d` but I'm not sure it's relevant since the mover is about swapping positions.